File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Interfaces/AcmeMessage.java
package com.github.grzesiek_galezowski.BastardInjection.Interfaces;

import com.github.grzesiek_galezowski.BastardInjection.Services.IAuthorization;

public interface AcmeMessage {
  void authorizeUsing(IAuthorization authorizationRules);

  void writeTo(DataDestination dataDestination);
}

File: CSharp/DependencyInjection/DependencyInjection/BastardInjection/_4_BastardInjection/Interfaces/AcmeMessage.cs
using BastardInjection._4_BastardInjection.Services;

namespace BastardInjection._4_BastardInjection.Interfaces
{
  public interface AcmeMessage
  {
    void AuthorizeUsing(IAuthorization authorizationRules);
    void WriteTo(DataDestination dataDestination);
  }
}